pkg/sql/distsql_physical_planner.go

```diff
@@ -3051,6 +3051,8 @@ func (dsp *DistSQLPlanner) createPlanForSetOp(
 					execinfrapb.PostProcessSpec{},
 					p.ResultTypes,
 				)
+			} else {
+				p.EnsureSingleStreamPerNode()
```

Comment:
say that this is necessary because otherwise we end up with result routers even when planning things on a single node and we can't fuse through those routers.

Reply:
Done, thanks.